### PR TITLE
[AutoMM] Fix a lightning warning

### LINKED ISSUE #2934: Deprecation warnings in multimodal
Multimodal logs many instances of deprecation warnings when fitting ([Example](https://auto.gluon.ai/dev/tutorials/multimodal/text_prediction/beginner_text.html#training))

```
/home/ci/opt/venv/lib/python3.8/site-packages/pytorch_lightning/trainer/connectors/accelerator_connector.py:585: LightningDeprecationWarning: The Trainer argument auto_select_gpus has been deprecated in v1.9.0 and will be removed in v2.0.0. Please use the function pytorch_lightning.accelerators.find_usable_cuda_devices instead.
  rank_zero_deprecation(
```

This should be fixed for v0.7 release.

### PR DESCRIPTION
**Pull request by zhiqiangdon**
*Issue #, if available:*
resolves #2934 

*Description of changes:*
Refer to PR: https://github.com/Lightning-AI/lightning/pull/16147

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: multimodal/src/autogluon/multimodal/utils/environment.py [hunk 1]
```diff
@@ -6,6 +6,7 @@
 from typing import Dict, List, Optional, Tuple, Union
 
 import torch
+from pytorch_lightning.accelerators import find_usable_cuda_devices
```
Comment (sxjscience) [inline multimodal/src/autogluon/multimodal/utils/environment.py:9]: Seems that it has been added very recently, and is part of `lightning_lite`:

```
from lightning_lite.accelerators.cuda import find_usable_cuda_devices
```

File: multimodal/src/autogluon/multimodal/predictor.py [hunk 2]
```diff
@@ -1342,10 +1343,11 @@ def _fit(
         with apply_log_filter(log_filter):
```
Comment (sxjscience) [inline multimodal/src/autogluon/multimodal/predictor.py:1343]: Is it possible for us to filter the warning here?

File: multimodal/src/autogluon/multimodal/utils/environment.py (hunk 1 above)
Comment (sxjscience) [reply multimodal/src/autogluon/multimodal/utils/environment.py:9]: Do we need to update our version bound of lightning? https://github.com/autogluon/autogluon/blob/48e1ad051654054f82d033a2c2e3ecc28515bee9/multimodal/setup.py#L40

Comment (sxjscience) [reply multimodal/src/autogluon/multimodal/utils/environment.py:9]: I just checked. The API is only available for `lightning>=1.9.0`:

- 1.9.0: https://github.com/Lightning-AI/lightning/blob/1.9.0/src/pytorch_lightning/accelerators/__init__.py (Has `find_usable_cuda_devices`)
- 1.8.6: https://github.com/Lightning-AI/lightning/blob/1.8.6/src/pytorch_lightning/accelerators/__init__.py (Does not have `find_usable_cuda_devices`)

Comment (tonyhoo) [reply multimodal/src/autogluon/multimodal/utils/environment.py:9]: If we need to use this function from lightning, we have to update the version lower bound

Comment (zhiqiangdon) [reply multimodal/src/autogluon/multimodal/utils/environment.py:9]: Lower bound is updated to 1.9.0.

File: multimodal/src/autogluon/multimodal/predictor.py (hunk 2 above)
Comment (zhiqiangdon) [reply multimodal/src/autogluon/multimodal/predictor.py:1343]: They will remove this flag in the v1.10.0, which I guess will come soon.

File: multimodal/src/autogluon/multimodal/utils/environment.py [hunk 3]
```diff
@@ -274,3 +275,31 @@ def check_if_packages_installed(problem_type: str):
...
+
+    Returns
+    -------
+    The available devices.
+    """
+    if num_gpus > 0 and not use_ray_lightning:  # ray lightning requires not specifying gpus
+        if auto_select_gpus:
+            devices = find_usable_cuda_devices(num_gpus)
+        else:
+            devices = num_gpus
```
[the remark this answers is not in the document]
Comment (zhiqiangdon) [reply multimodal/src/autogluon/multimodal/utils/environment.py:301]: If auto_select_gpus is False, then we use the lightning trainer to handle this, which is lightning's the default setting.

**Review by tonyhoo — APPROVED** [review]
LGTM!

**Review by sxjscience — APPROVED** [review]
LGTM!

**Review by Innixma — APPROVED** [review]
Thanks for fixing the warnings!! Much appreciated, LGTM!

**Pull request merged**